docs/authorization/globus_app/apps.rst

```diff
@@ -0,0 +1,89 @@
+.. _globus_apps:
+
+.. currentmodule:: globus_sdk
+
+GlobusApps
+==========
+
+
+.. warning::
+
+    Currently ``GlobusApp`` only supports scripting-based use cases.
+
+    Hosted applications (i.e. web servers) should see :ref:`globus_authorizers` instead.
+
+
+:class:`GlobusApp` is a high level construct designed to simplify the process of
+of authenticating for Globus service clients.
+
+A ``GlobusApp`` will always require a programmatic entity called a "client". Depending
+on the type of ``GlobusApp`` in use, the client will play different roles in the
+request flow.
+
+*   A :class:`UserApp` drives interaction starting with a browser based login
+    flow. With this type of ``GlobusApp``, newly created resources (collections,
+    flows, etc.) are owned by the human user who has logged in (and thus granted the
+    app authorization tokens). Similarly resource access is evaluated based on that
+    user's permissions.
+
+    A :class:`UserApp` should be used in cases where a script simplifies a process
+    for a user, but the user is ultimately responsible for the actions taken by the
+    script. The `Globus CLI <https://docs.globus.org/cli/>`_ is a good example of
+    one such use case.
+
+    A :class:`UserApp` **should use a "native"-type client**. "Confidential"-type
+    clients are allowed, but prove more difficult to configure.
```

Comment:
```suggestion
    A :class:`UserApp` **should use a "native" client**. "Confidential"
    clients are allowed, but prove more difficult to configure.
```

Reply:
Follow-up question: Is there _ever_ a compelling reason to use a confidential client in this case? (i.e., if there is a situation where that's actually the correct solution, it _might_ be useful to say something like "Confidential clients are allowed but should only be used if you need to ...")

Reply:
My short answer here is "yes".

There are only two specific cases I know of: 
- a hosted client (e.g., a Django app)
- a templated client (e.g., actual cli implementation) 

I still wonder a little if we should split UserApp into multiple classes for the different cases, but I figured that we can add subclasses in the future if we want.

Reply:
This concept is reflected in the table now.